### PR TITLE
Toggle Option: Ability for civs to revive others

### PR DESCRIPTION
**Pull request by SimZor**
Resolves No Issue
#### Changes proposed in this pull request:
- Let people that use the Framework decide if they want civs the ability to use defibrillator and revive people
- Toggleable option in order to get civs to use defib

### CODE REVIEW

**Comment by SimZor** [comment]
Status on this?

**Comment by Jawshy** [comment]
@SimZor, waiting for @BoGuu to review.

**Comment by DarkSilencerZA** [comment]
@SimZor might take @BoGuu a while. xD

**Comment by SimZor** [comment]
@DarkSilencerZA kek right? :D

File: Altis_Life.Altis/core/functions/fn_actionKeyHandler.sqf [hunk 1]
```diff
@@ -81,9 +81,9 @@ life_action_inUse = true;
 };
 
 //Check if it's a dead body.
-if (_curObject isKindOf "Man" && !(_curObject isKindOf "Animal") && {!alive _curObject} && !(_curObject getVariable ["Revive",false]) && {playerSide in [west,independent]}) exitWith {
```
Comment (BoGuu) [inline Altis_Life.Altis/core/functions/fn_actionKeyHandler.sqf]: East is coming!

!(side -> east)

File: Altis_Life.Altis/core/functions/fn_actionKeyHandler.sqf [hunk 2]
```diff
@@ -81,9 +81,9 @@ life_action_inUse = true;
...
+if (_curObject isKindOf "Man" && !(_curObject isKindOf "Animal") && {!alive _curObject} && !(_curObject getVariable ["Revive",false])) exitWith {
     //Hotfix code by ins0
-    if (((playerSide isEqualTo west && {(LIFE_SETTINGS(getNumber,"revive_cops") isEqualTo 1)}) || playerSide isEqualTo independent)) then {
+    if (((playerSide isEqualTo west && {(LIFE_SETTINGS(getNumber,"revive_cops") isEqualTo 1)}) || (playerSide isEqualTo civilian && {(LIFE_SETTINGS(getNumber,"revive_civ") isEqualTo 1)}) || playerSide isEqualTo independent)) then {
```
Comment (BoGuu) [inline Altis_Life.Altis/core/functions/fn_actionKeyHandler.sqf]: Lazy eval on the inde part :)

**Comment by BoGuu** [comment]
I'm not really into civs getting money to revive (trying to imagine an in game situation, kind of like a medic in a gang?). Thoughts everyone?

**Comment by imKLM** [comment]
yeah have a feeling it would be abused and too hard to manage (rule wise). would leave money out

**Comment by SimZor** [comment]
Sorry for late response. Currently on vacation in Florida. I'll try looking over this tomorrow.

plz no close PR ;)

**Comment by SimZor** [comment]
Everything should be in now that you requested @BoGuu

**Comment by SimZor** [comment]
I updated it @BoGuu I think I had a bit of a brain freeze and a bit tired at night when I wrote west :D

File: Altis_Life.Altis/core/medical/fn_revivePlayer.sqf [hunk 3]
```diff
@@ -66,7 +66,7 @@ if (player getVariable ["restrained",false]) exitWith {life_action_inUse = false
 if (!isNil "_badDistance") exitWith {titleText[localize "STR_Medic_TooFar","PLAIN"]; life_action_inUse = false;};
 if (life_interrupted) exitWith {life_interrupted = false; titleText[localize "STR_NOTF_ActionCancel","PLAIN"]; life_action_inUse = false;};
 
-if (playerSide isEqualTo west) then {
+if ( playerSide isEqualTo independent ) then {
```
Comment (Jawshy) [inline Altis_Life.Altis/core/medical/fn_revivePlayer.sqf]: :japanese_goblin: `( playerSide`

**Comment by NeFShARk** [comment]
Idk if this is relevant or not, its just an idea! I think that we need a check for "online medics", if there is a certain number of medics online at the moment then the player will not be able to revive others and he will receive a message saying that he needs to call a medic.

Also the "minimal number of medics" for this check should be add to the config so anyone using the framework will be able to set how many medics needs to be online for this check to be disabled or enabled, just like the minimal number of cops for players to rob the federal reserve.

I think this is important because the ability for civs to revive others could make the "medics role" somewhat irrelevant since most players that play altis life usually play with other people in groups, what i mean by this, is that the civs will choose to revive each other instead of calling a medic even if there are a lot of medics online. 

Off course that this feature(ability for civs to revive others) is important! It sucks when there is no medics online and someone dies and they want to be revived because they don't want to loose their gear, when this happens, usually these players keep bothering the admins about it.

**Comment by SimZor** [comment]
@NeFShARk Good suggestion! I agree with it, however, keep in mind that you can toggle reviving for civs and cops on/off. I can add this, if I get @BoGuu's go ahaed!

**Comment by NeFShARk** [comment]
@SimZor Yes, i understand.

I also think that if a player request a medic using the "request medic button" then he can't be revived by another civ, in others words, it's another check to prevent a player to revive another player if this player requested a medic, each one of them(the civs, the dead one and the one trying to revive him) would then receive a message saying "You can't be revived by another civ because you requested a medic" and "You can't revive this civ because he requested a medic".

Another suggestion, idk, maybe this one is not valid and it would take to much to be done but anyway...

The suggestion is that along with the medics online check, you also add a distance check, if there aren't any medics in a X distance from the target(in this case the dead civ), then another civ can revive him, if there are medics in this X distance from the target, then he needs to call a medic! Again the civs, each one of them would receive a message like "You can't revive this civ, there medics in range, he needs to call one!" and "You can't be revived by another civ because there are medics in range, call one!".

/\ I suggested this because sometimes is a pain in the ass to wait for a medic to arrive when there are medics online because they are to far from the target, altis map and tanoa are pretty big! It's even worse when you are playing on a server that doesn't allow every medic to use helicopters.

Idk, what you guys think? Maybe it's to much hehe :P Anyway these are suggestions only :P

**Comment by SimZor** [comment]
I think these are good suggestions @Nefshark, however as you mentioned: putting all that in would maybe be to much.

I think we should focus on one of the different alternatives you mentioned. 

Also: allowing civs to revive players without them requesting medic could be quite abused IMO.

File: Altis_Life.Altis/core/functions/fn_actionKeyHandler.sqf [hunk 4]
```diff
@@ -81,9 +81,9 @@ life_action_inUse = true;
 };
 
 //Check if it's a dead body.
-if (_curObject isKindOf "Man" && !(_curObject isKindOf "Animal") && {!alive _curObject} && !(_curObject getVariable ["Revive",false]) && {playerSide in [west,independent]}) exitWith {
+if (_curObject isKindOf "Man" && !(_curObject isKindOf "Animal") && {!alive _curObject} && !(_curObject getVariable ["Revive",false])) exitWith {
     //Hotfix code by ins0
-    if (((playerSide isEqualTo west && {(LIFE_SETTINGS(getNumber,"revive_cops") isEqualTo 1)}) || playerSide isEqualTo independent)) then {
+    if (((playerSide isEqualTo west && {(LIFE_SETTINGS(getNumber,"revive_cops") isEqualTo 1)}) || (playerSide isEqualTo civilian && {(LIFE_SETTINGS(getNumber,"revive_civ") isEqualTo 1)}) || (playerSide isEqualTo east && {(LIFE_SETTINGS(getNumber,"revive_east") isEqualTo 1)}) || playerSide isEqualTo independent)) then {
```
Comment (BoGuu) [inline Altis_Life.Altis/core/functions/fn_actionKeyHandler.sqf]: Lazy eval :)

File: Altis_Life.Altis/core/medical/fn_revivePlayer.sqf [hunk 5]
```diff
@@ -66,13 +66,19 @@ if (player getVariable ["restrained",false]) exitWith {life_action_inUse = false
...
+    BANK = BANK + _reviveCost;
+};
+
 [1] call SOCK_fnc_updatePartial;
 
 life_action_inUse = false;
 _target setVariable ["Revive",true,true];
 [profileName] remoteExecCall ["life_fnc_revived",_target];
-titleText[format [localize "STR_Medic_RevivePayReceive",_targetName,[_reviveCost] call life_fnc_numberText],"PLAIN"];
+
```
Comment (BoGuu) [inline Altis_Life.Altis/core/medical/fn_revivePlayer.sqf]: Can combine both if sections

**Comment by NeFShARk** [comment]
@BoGuu, @Jawshy and anyone else......

What do you guys think about the suggestions that i've made above?

**Comment by BoGuu** [comment]
@NeFShARk 

Make an issue with any suggestions and everyone can chip in there! :)

**Comment by NeFShARk** [comment]
@BoGuu 

Done :)

**Comment by Jawshy** [comment]
LGTM

**Comment by SimZor** [comment]
Great!

**Comment by SimZor** [comment]
Status?

**Comment by BoGuu** [comment]
LGTM

**Pull request merged**